addons/l10n_it_edi/models/account_edi_format.py

```diff
@@ -524,57 +494,51 @@ def _import_fattura_pa(self, tree, invoice):
-                                if total_discount_amount != 0:
-                                    discount = {}
-                                    discount["seq"] = invoice_line_form.sequence + 1
-
-                                    if total_discount_amount < 0:
-                                        discount["name"] = _('DISCOUNT: %s', invoice_line_form.name)
-                                    else:
-                                        discount["name"] = _('EXTRA CHARGE: %s', invoice_line_form.name)
-                                    discount["amount"] = total_discount_amount
-                                    discount["tax"] = []
-                                    for tax in invoice_line_form.tax_ids:
-                                        discount["tax"].append(tax)
-                                    discount_list.append(discount)
-                            invoice_line_form.discount = (1 - total_discount_percentage) * 100
-
-                # Apply amount discount.
-                for discount in discount_list:
-                    with invoice_form.invoice_line_ids.new() as invoice_line_form_discount:
-                        invoice_line_form_discount.tax_ids.clear()
-                        invoice_line_form_discount.sequence = discount["seq"]
-                        invoice_line_form_discount.name = discount["name"]
-                        invoice_line_form_discount.price_unit = discount["amount"]
+                            # Discount in cascade summarized in 1 percentage
+                            discount_elements = element.xpath('.//ScontoMaggiorazione')
+                            if discount_elements:
+                                discount_element = discount_elements[0]
+                                discount_percentage = discount_element.xpath('.//Percentuale')
+                                if discount_percentage and len(discount_elements) == 1:
+                                    discount_type = discount_element.xpath('.//Tipo')
+                                    discount_sign = 1 
+                                    if discount_type and discount_type[0].text == 'MG':
+                                        discount_sign = -1
+                                    invoice_line_form.discount = discount_sign * float(discount_percentage[0].text)
+                                else:
+                                    total = float(element.xpath('.//PrezzoTotale')[0].text)
+                                    discount = 100 - (100 * total) / (invoice_line_form.quantity * invoice_line_form.price_unit)
+                                    invoice_line_form.discount = discount
+
+
+                # Global discount summarized in 1 amount
+                discount_elements = body_tree.xpath('.//DatiGeneraliDocumento/ScontoMaggiorazione')
+                if discount_elements:
+                    taxable_amount = sum([float(amount.text) for amount in body_tree.xpath('.//ImponibileImporto')])
```

Comment:
This is weird and we should have it in the invoice already. (invoice.amount_untaxed or something)  No need to get it from the XML itself normally.  Trying to understand myself, the global discount should not have an impact on the tax. (https://fatturapa.supermercato.it/documenti/EFFETTI_SU_GESTIONE_SCONTI.pdf)  

So we just need to take the untaxed amount on the invoice.